### PR TITLE
Use stylesheet’s base URL and ns prefixes in CSSOM insert/appendRule.

### PR DESCRIPTION
**Pull request by SimonSapin**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14395)

### CODE REVIEW

File: components/script/dom/cssrule.rs [hunk 1]
```diff
@@ -17,27 +17,34 @@ use dom::cssstylerule::CSSStyleRule;
...
 #[dom_struct]
 pub struct CSSRule {
     reflector_: Reflector,
-    parent: MutNullableHeap<JS<CSSStyleSheet>>,
+    parent_stylesheet: JS<CSSStyleSheet>,
+
+    /// Whether the parentStyleSheet attribute should return null.
+    /// We keep parent_stylesheet in that case because insertRule needs it
+    /// for the stylesheet’s base URL and namespace prefixes.
+    parent_stylesheet_removed: Cell<bool>,
```
Comment (Manishearth) [inline components/script/dom/cssrule.rs]: What does the spec say about this case? 


... does it even matter?

**Review by Manishearth — APPROVED** [review]
r+

File: components/script/dom/cssrule.rs (hunk 1 above)
Comment (Manishearth) [inline components/script/dom/cssrule.rs]: Could you file an issue or a PR to make it explicit?

Comment (SimonSapin) [inline components/script/dom/cssrule.rs]: idk, seems obvious that relative URLs shouldn’t suddenly change meaning.

**Pull request merged**